.github/workflows/rust.yml

```diff
@@ -46,36 +46,29 @@ jobs:
           # and thus do not depend on the OS, arch nor rust version.
           path: /github/home/.cargo
           key: cargo-cache-
-      - name: Cache Rust dependencies
```

Comment:
I have never been convinced this caching is particularly effective so I am going to test what happens when I remove it...

Reply:
I found this https://github.com/marketplace/actions/rust-cache which does perform some heuristics and cleaning.

Reply:
Maybe what's wrong here is that the `.cargo` die is not cached?

https://doc.rust-lang.org/cargo/guide/cargo-home.html#caching-the-cargo-home-in-ci

Reply:
```
This cache is automatically keyed by:

the github [job_id](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_id),
the rustc release / host / hash,
the value of some compiler-specific environment variables (eg. RUSTFLAGS, etc), and
a hash of all Cargo.lock / Cargo.toml files found anywhere in the repository (if present).
a hash of all rust-toolchain / rust-toolchain.toml files in the root of the repository (if present).
```

Seems particularly relevant to why the basic cache wasn't working for us